File: lib/src/main/java/com/dhy/retrofitrxutil/IErrorHandler.java
package com.dhy.retrofitrxutil;

import android.app.Activity;
import android.app.Dialog;
import android.content.Context;
import android.support.annotation.NonNull;

public interface IErrorHandler {
    void onError(@NonNull IObserverX observer, @NonNull Throwable e);

    Dialog showDialog(@NonNull Context context, @NonNull String msg);

    @NonNull
    IError parseError(@NonNull Throwable e);

    /**
     * if failed then logout. Should not be LoginActivity
     */
    boolean isAuthorizeFailed(@NonNull Activity activity, int errorCode);

    void onLogout(@NonNull Context context);

    void onActivityError(@NonNull Activity activity, @NonNull IError error);

    void onBackgroundError(@NonNull Context context, @NonNull IError error, @NonNull Throwable e);

    boolean isDebug();
}

File: lib/src/main/java/com/dhy/retrofitrxutil/IErrorHandler.kt
package com.dhy.retrofitrxutil

import android.app.Activity
import android.app.Dialog
import android.content.Context

interface IErrorHandler {
    fun onError(observer: IObserverX, e: Throwable)
    fun showDialog(context: Context, msg: String): Dialog?
    fun parseError(e: Throwable): IError
    /**
     * if failed then logout. Should not be LoginActivity
     */
    fun isAuthorizeFailed(activity: Activity, errorCode: Int): Boolean

    fun onLogout(context: Context)
    fun onActivityError(activity: Activity, error: IError)
    fun onBackgroundError(context: Context, error: IError, e: Throwable)
    fun isDebug(): Boolean
}